### PR TITLE
Add IconTabLayout tests

### PR DESCRIPTION
**Pull request by arturdm**
This PR also adds a way to test Android library classes that use Android resources from other libraries.

### CODE REVIEW

File: lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    configureMock(pagerAdapter);
+    configureMock(layoutPagerAdapter);
+    given(layoutPagerAdapter.getPageIcon(0)).willReturn(mock(Drawable.class));
+    given(layoutPagerAdapter.getPageIcon(1)).willReturn(mock(Drawable.class));
+  }
+
+  @Test public void shouldSetTabsFromPagerAdapterInstance() {
+    // given
+    IconTabLayout iconTabLayout = new IconTabLayout(RuntimeEnvironment.application);
+    given(pagerAdapter.getCount()).willReturn(2);
```
Comment (tomrozb) [inline lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java]: Is this line necessary?

Comment (arturdm) [inline lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java]: Leftover after refactoring, I'll remove that.

File: lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  private void configureMock(@NonNull PagerAdapter pagerAdapter) {
+    given(pagerAdapter.getCount()).willReturn(2);
+    given(pagerAdapter.getPageTitle(0)).willReturn("barbers");
+    given(pagerAdapter.getPageTitle(1)).willReturn("lumberjacks");
+  }
+
+  @NonNull private AttributeSet createAttributeSetWithPageTitlesVisible(@NonNull Context context,
+      boolean pageTitlesVisible) {
+    return new RoboAttributeSet(Lists.newArrayList(
```
Comment (tomrozb) [inline lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java]: Didn't know assertj has Lists similar to Guava. Maybe I should switch to static factory methods in SignInButtonTest

**Comment by tomrozb** [comment]
+1

File: lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java (hunk 2 above)
Comment (arturdm) [inline lib-android/src/test/java/com/devinci/lib/view/IconTabLayoutTest.java]: Or maybe I should have refactored your code instead :).

**Comment by arturdm** [comment]
If I run tasks in this order I get errors:

```
$ ./gradlew cC jacocoTestReport

...

:lib-android:compileDebugAndroidSupportDesign
error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider dagger.internal.codegen.ComponentProcessor not found
:lib-android:compileDebugAndroidSupportDesign FAILED
:lib-android:copyDebugAndroidSupportV7Appcompat UP-TO-DATE
:lib-android:compileDebugAndroidSupportV7Appcompat
error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider dagger.internal.codegen.ComponentProcessor not found
:lib-android:compileDebugAndroidSupportV7Appcompat FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lib-android:compileDebugAndroidSupportDesign'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lib-android:compileDebugAndroidSupportV7Appcompat'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
==============================================================================

BUILD FAILED
```

**Comment by tomrozb** [comment]
+1 once again

File: lib-android/src/test/java/com/devinci/lib/AttributeSetUtils.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import android.support.annotation.NonNull;
+import android.util.AttributeSet;
+import org.robolectric.res.Attribute;
+import org.robolectric.shadows.RoboAttributeSet;
+
+import static org.assertj.core.util.Lists.newArrayList;
+import static org.assertj.core.util.Preconditions.checkNotNull;
+import static org.robolectric.Shadows.shadowOf;
+
+public final class AttributeSetUtils {
```
Comment (tomrozb) [inline lib-android/src/test/java/com/devinci/lib/AttributeSetUtils.java:13]: We should think about `lib-android-test-robolectric` module. This class  may be useful in other apps.

Comment (arturdm) [inline lib-android/src/test/java/com/devinci/lib/AttributeSetUtils.java:13]: :+1:

**Pull request merged**